needs more work





From: Byron Bradley <byron.bbradley@gmail.com>
Date: Fri, 8 Feb 2008 02:20:35 +0000 (+0000)
Subject: sata_mv: problems using it as a platform_driver
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Fnico%2Forion.git;a=commitdiff_plain;h=1c62b99e64caa0e96fe9d6d3a4f75f54db8f0c7f

sata_mv: problems using it as a platform_driver

On Fri, 8 Feb 2008, Byron Bradley wrote:

> In mv_platform_probe() host->iomap is set to NULL but it is dereferenced
> in mv_start_dma(), I'm not sure what the fix for this is.

The following patch makes this driver work although I have no idea if this
is the correct thing to do. Will this be OK for other devices using this
driver? The pointer only works because it is dereferenced as
iomap[MV_PRIMARY_BAR] and MV_PRIMARY_BAR = 0, will another offset ever be
used in the future?
---

diff --git a/drivers/ata/sata_mv.c b/drivers/ata/sata_mv.c
index 8b79f1e..ef10578 100644
--- a/drivers/ata/sata_mv.c
+++ b/drivers/ata/sata_mv.c
@@ -2948,8 +2948,8 @@ static int mv_platform_probe(struct platform_device *pdev)
 	host->private_data = hpriv;
 	hpriv->n_ports = n_ports;
 
-	host->iomap = NULL;
 	hpriv->base = ioremap(res->start, res->end - res->start + 1);
+	host->iomap = &hpriv->base;
 	hpriv->base -= MV_SATAHC0_REG_BASE;
 
 	rc = mv_create_dma_pools(hpriv, &pdev->dev);
